src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,249 @@
+            String [] splitInput = input.split(" ");
+
+            // list commands duke to list all the tasks stored and their completion status
+            // try at the start cos of the errors possibly
+            try {
+                switch (splitInput[0]) {
+                    case "list":
+                        System.out.println("Here are the tasks in your list:");
+                        for (int i = 0; i < tasks.size(); i++) {
+                            System.out.println(i + 1 + ". " + tasks.get(i));
+                        }
+                        System.out.println();
+                        break;
+
+                    case "mark":
+                        // mark x commands duke to mark the corresponding task as completed
+                        numericString = input.substring(input.indexOf(" ") + 1);
+                        markedNum = Integer.parseInt(numericString) - 1;
+                        tasks.get(markedNum).setStatus(Boolean.TRUE);
+                        //print to CLI
+                        System.out.println("Nice! I've marked this task as done: ");
+                        System.out.println("  " + tasks.get(markedNum) + "\n");
+                        //save
+                        writeToFile(filePath,tasks);
+                        break;
+
+                    case "unmark":
+                        // unmark x commands duke to mark the corresponding task as uncompleted
+                        // Exceptions could occur
+                        numericString = input.substring(input.indexOf(" ") + 1);
+                        markedNum = Integer.parseInt(numericString) - 1;
+                        tasks.get(markedNum).setStatus(Boolean.FALSE);
+                        //print to CLI
+                        System.out.println("Oh no :( I've marked it as uncompleted: ");
+                        System.out.println("  " + tasks.get(markedNum) + "\n");
+                        //save
+                        writeToFile(filePath,tasks);
+                        break;
+
+                    case "todo":
+                        if (splitInput.length == 1) {
+                            throw new EmptyDescriptionException();
+                        }
+                        description = input.substring(input.indexOf(" ") + 1);
+                        Task td = new Todo(description);
+
+                        // print to CLI
+                        taskPrint(tasks, td);
+                        //save
+                        appendToFile(filePath,td);
+                        break;
+
+                    case "deadline":
+                        if (splitInput.length == 1) {
+                            throw new EmptyDescriptionException();
+                        }
+                        description = input.substring(input.indexOf(" ") + 1, input.indexOf(" /by "));
+                        String by = input.substring(input.indexOf("/by ") + 4);
```

Comment:
Might want to refactor this kind of parsing into a separate method or at least something that doesn't use explicit indices.